### PR TITLE
ocp4_workload_rhel_worker: new workload

### PR DESCRIPTION
**Pull request by newgoliath**
##### SUMMARY

This new ocp4_workload role is (I hope) useful with any OCP4 cluster config (or BYO cluster).

It deploys any number of RHEL8 instances to be used as worker nodes.   It does not run the playbooks to install OpenShift on the RHEL nodes, or automate their joining the cluster.  That's for the student to do.  Samples code of the steps necessary to setup OpenShift on the RHEL nodes and have them join the cluster are in the workload.yml comments.

It deploys the RHEL instances into the VPC of the existing cluster in the sandbox.

It sets up all AWS permissions necessary for the new RHEL nodes to join the cluster.

It updates the bastion host with a checkout of the openshift-ansible project and a list of RHEL_workers.txt to be added to an ansible inventory.

The user must run the openshift-ansible playbooks from the Bastion in order to install OpenShift on the RHEL nodes, and the playbooks will approve the CSRs and have the new RHEL workers join the cluster.
 

##### ISSUE TYPE

- New role Pull Request

##### COMPONENT NAME

roles_ocp_workloads/ocp4_workload_rhel_worker

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/readme.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+* This role installs playbooks and shell scripts to deploy a RHEL8 Worker for AWS based deployments.
+
+Once this role has been executed there is a directory `/opt/rhel_worker` with the necessary files to deploy a rhel_worker.
+If `ocp4_workloads_rhel_worker_deploy_worker` is set, this workload will deploy one worker.
+You can deploy subsequent workers with the playbook.
+SSH into the bastion, change to the `/opt/rhel_worker` directory and run `./setup_rhel_worker.sh`.
+This script will execute the playbook to deploy a rhel_worker into the region specified.
+
+Currently only `eu-central-1`, `us-east-1` and `us-east-2` are supported. This is because only for these three regions the AMI for the RHEL 8 image is specified.
```
Comment (fridim) [inline ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/readme.adoc:13]: i see you're using image name and owner to find the image.

Isn't the image available in other regions?

File: ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/tasks/workload.yml [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+  - name: set a fact for the instance tags
+    set_fact:
+      instance_tags: >-
+        {
+          "type": "RHEL_worker",
+          "guid": "{{ guid }}",
+          "cluster_name": "{{ cluster_name }}",
+          "kubernetes.io/cluster/{{ cluster_name }}": "{{ ocp4_workload_rhel_worker_ownership }}"
+        }
```
Comment (fridim) [inline ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/tasks/workload.yml:189]: that's not enough.

You need to merge `hostvars.localhost.cloud_tags_final` on top of this dictionary to have every tags (uuid, platform, etc). Some are mandatory for cost reporting.

File: ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/tasks/workload.yml [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+      region: "{{ aws_region }}"
+      key_name: "opentlc_admin_backdoor"
+      vpc_subnet_id: "{{ ec2_vpc_subnet_ids.subnets[0].subnet_id }}"
+      instance_type: m5.4xlarge
+      group_id: "{{ sg_RHEL_worker.group_id }}"
+      image: "{{ __rhel_ami.images[0].image_id }}"
+      assign_public_ip: true
+      exact_count: "{{ ocp4_workload_rhel_worker_exact_count }}"
+      count_tag: "{{ instance_tags }}"
+      instance_tags: "{{ instance_tags }}"
```
Comment (fridim) [inline ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/tasks/workload.yml:202]:
```yaml
      instances_tags: "{{ instance_tags | combine(hostvars.localhost.cloud_tags_final) }}"
```

File: ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/readme.adoc [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+*** Debug task will print out: `post_workload Tasks completed successfully.`
+
+** Tasks: link:./tasks/remove_workload.yml[remove_workload.yml] - Used to delete the workload
+*** This role doesn't do anything here
+*** Debug task will print out: `remove_workload Tasks completed successfully.`
+
+== Review the defaults variable file
+
+* This file link:./defaults/main.yml[./defaults/main.yml] contains all the variables you need to define to control the deployment of your workload.
+* The variable *ocp_username* is mandatory to assign the workload to the correct OpenShift user.
```
Comment (fridim) [inline ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/readme.adoc:33]: i don't see `ocp_username` as a mandatory variable. the variable is not used except in a `debug` task?

File: ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/readme.adoc (hunk 1 above)
Comment (newgoliath) [reply ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/readme.adoc:13]: readme updated

File: ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/readme.adoc (hunk 4 above)
Comment (newgoliath) [reply ansible/roles_ocp_workloads/ocp4_workload_rhel_worker/readme.adoc:33]: readme.adoc updated
Cluster admin is assumed

**Review by fridim — APPROVED** [review]

**Pull request merged**